File: app/channels/image_channel.rb
class ImageChannel < ApplicationCable::Channel
  def subscribed
    stream_from "image_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def received(data)
    ActionCable.server.broadcast "image_channel", message: data["message"]
  end
end

File: app/javascript/channels/image_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("ImageChannel", {
  connected() {
    console.log("Connected to ImageChannel");
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
  },

  send(message) {
    this.perform('received', { message: message });
  }
});
